### ISSUE TITLE
Is Sampyl still being developed/maintained?

### ISSUE THREAD

**Issue by Gabriel-p** [mcleonard/sampyl#18]
Hi, I'm looking for a simple to use Bayesian MCMC sampler fully Python based. I've been trying `emcee` and now I've found Samply, which looks really good.

Before attempting to implement it though, I'd like to know if it is still being developed or at least maintained. For what I can see the latest commit is 1.5 years old with no active branches, and the opened issues are ~2 years old with no apparent progress in either of them.

Thanks.

**Comment by twiecki** [comment]
Have you looked at PyMC3?

**Comment by ipashchenko** [comment]
Here the author explains why he has implemented sampyl and what is a
problem with pymc3:
http://matatat.org/ab-testing-with-sampyl.html


3 нояб. 2017 г. 1:33 ДП пользователь "Thomas Wiecki" <
notifications@github.com> написал:

**Comment by Gabriel-p** [comment]
@twiecki I have, but as @ipashchenko comments it depends on an annoying syntax to write your models. It's the same issue with PyStan, and that's why Sampyl looks so interesting.

**Comment by twiecki** [comment]
Yeah, theano makes things tricky but I'm always surprised that people would rather code their own likelihoods and model logps.

**Comment by ipashchenko** [comment]
@Gabriel-p , there's nice and easy to use python implementation of the nested sampling [here](https://github.com/kbarbary/nestle) by @kbarbary. So if you are not sticked to MCMC and want to get evidence estimates for e.g. model comparison - just check it.

**Comment by Gabriel-p** [comment]
@twiecki In my case my likelihood is *already* written, and it is not trivial. I'd have to invest the time into learning Theano (or PyStan's syntax) before even attempting to get a glimpse of how adequate the package is for my particular issue.

@ipashchenko I'll definitely look into it. Thank you.

**Comment by madanh** [comment]
@twiecki For me it's the opposite. I see the necessity to learn a domain specific language and a limited set of building blocks as the (annoying) price to pay for automatic differentiation and the speed-ups it brings. Perhaps this difference in views is that for people formally trained in statistic navigating the zoo of available distributions is easy while for people who learn Bayesian probability and MCMC to make inferences in their specific problems it is quite problematic.

**Comment by Gabriel-p** [comment]
Now even Theano is being discontinued too: https://groups.google.com/forum/#!msg/theano-users/7Poq8BZutbY/rNCIfvAEAwAJ

**Comment by madanh** [comment]
@Gabriel-p 
Did you try using sampyl as it is?
I cloned it and and tried the examples from docs. Those wouldn't run due to a bug that I managed to fix/work around and now the examples are working fine. So if you want to try this now I can fork and push what I have. Will definitely play a bit more with this.

**Comment by twiecki** [comment]
Just submit a PR to this repo. Thanks!

**Comment by twiecki** [comment]
I have a close line to @colcarroll and will make sure it will get merged :)

**Comment by twiecki** [comment]
I don't and actually confused this repo with "sampled" so disregard my earlier comment. Still think a PR couldn't hurt though :)

**Comment by ipashchenko** [comment]
@madanh , what is the problem? [Check here](https://stackoverflow.com/questions/6022302/how-to-apply-unmerged-upstream-pull-requests-from-other-forks-into-my-fork)

**Comment by madanh** [comment]
@ipashchenko I could not quite understand your message. Care to elaborate? 
@twiecki Happens :) 
Anyway, here's the PR. https://github.com/mcleonard/sampyl/pull/19

**Comment by Gabriel-p** [comment]
@madanh I have not used sampyl yet, I found while scavenging for Python MCMCs and thought it looked interesting, which is why I am interested in knowing about its development.

**Comment by madanh** [comment]
@Gabriel-p Apparently is is not maintained, so it is up to us to pick up the banner.

**Comment by madanh** [comment]
Good to have you back and nice work BTW. Do you have something to apply sampyl to now. It's my impression is that it's hard to find motivation to maintain something that does not help solve ones own everyday problems (unless one gets paid for this).

**Comment by zoj613** [comment]
any updates on the development of this project?

**Comment by jeremysanders** [comment]
Would it be possible to give someone else write access to at least fix the obvious bugs?
